app/models/course/assessment.rb

```diff
@@ -3,7 +3,10 @@
 # An assessment is a collection of questions that can be asked.
 class Course::Assessment < ActiveRecord::Base
   acts_as_lesson_plan_item
+  has_one_folder
 
+  after_initialize :build_initial_folder, if: :new_record?
+  before_validation :assign_folder_attributes
```

Comment:
I'm curious if there's a way to make this into some kind of DSL for has_one_folder, so we won't have to keep writing these two.